iam.tf

```diff
@@ -1,33 +1,49 @@
 locals {
-  iam_map = { for iam in var.iam : iam.role => iam }
+  # filter all objects that define a single role
+  iam_role = [for iam in var.iam : iam if can(iam.role)]
+
+  # filter all objects that define multiple roles and expand them to single roles
+  iam_roles = flatten([for iam in var.iam :
+    [for role in iam.roles : merge(iam, { role = role })] if can(iam.roles)
+  ])
```

Comment:
this also adds `iam.roles` support which needs to be documented.

Reply:
you mean [this](https://github.com/mineiros-io/terraform-google-pubsub-subscription/blob/537fdc48944316397044d30ff965b8544e61a6f6/README.md?plain=1#L321)?

Reply:
that's `role` but this also adds support for `roles` as described in another module here: https://github.com/mineiros-io/terraform-google-kms-key-ring/blob/589f4691c1902ab576c8d2209717c182a6e261b2/README.tfdoc.hcl#L282